core/src/Dscp/Core/Foundation/Witness.hs

```diff
@@ -236,26 +242,50 @@ newtype GTxId = GTxId (Hash GTx)
 
 -- | Compute tx id.
 toGTxId :: (Serialise Tx, Serialise PublicationTx) => GTx -> GTxId
-toGTxId (GMoneyTx tx) = GTxId . unsafeCastHash . toTxId $ tx
+toGTxId (GMoneyTx tx)       = GTxId . unsafeCastHash . toTxId $ tx
 toGTxId (GPublicationTx tx) = GTxId . unsafeCastHash . toPtxId $ tx
 
-data GTxWitnessed
+data WithSize a = WithSize
```

Comment:
Please stop putting utility datatypes into the `Witness` module, it's designed to carry only the logic-related types of witness, not serialisation wrappers.